### PR TITLE
[WebProfilerBundle][HttpKernel] Make FileLinkFormatter URL format generation lazy

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26755, #26754
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by tomtomau** [comment]
Will we have to wait around a month to get this fix in either 4.1 or 4.0.8? Was waiting for symfony/symfony#26568 in 4.0.7 but then this broke my functional tests. So just trying to work out whether to fork and apply this patch or just wait.

Cheers!

File: src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php [hunk 1]
```diff
@@ -53,6 +56,11 @@ public function load(array $configs, ContainerBuilder $container)
             $container->setParameter('web_profiler.debug_toolbar.intercept_redirects', $config['intercept_redirects']);
             $container->setParameter('web_profiler.debug_toolbar.mode', $config['toolbar'] ? WebDebugToolbarListener::ENABLED : WebDebugToolbarListener::DISABLED);
         }
+
+        if (Kernel::VERSION_ID >= 30408 || Kernel::VERSION_ID >= 40008) {
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php]: this needs to be `Kernel::VERSION_ID >= 30408 && Kernel::VERSION_ID < 40000 || Kernel::VERSION_ID >= 40008)`

Comment (nicolas-grekas) [reply src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php]: fixed in 946eefa

**Comment by enumag** [comment]
Can someone tell my why is `WebProfilerBundle` active in test env? What is it used for in tests?

**Comment by tomtomau** [comment]
We use it in a functional test to assert that an email was sent during a
request as there is a Swiftmailer integration into the Profiler

**Comment by andrerom** [comment]
@nicolas-grekas Any chance of getting this out soon? Having the following kind of errors reported with what I presume is this issue:
```
In Container.php line 297:
                                                                               
  Circular reference detected for service "routing.loader", path: "routing.lo  
  ader".   
```

**Pull request merged**